exec/java-exec/src/main/java/org/apache/drill/exec/store/dfs/FileSelection.java

```diff
@@ -190,22 +136,40 @@ private static String commonPath(FileStatus... paths) {
         break;
       }
     }
-    URI oneURI = paths[0].getPath().toUri();
+    URI oneURI = statuses.get(0).getPath().toUri();
     return new Path(oneURI.getScheme(), oneURI.getAuthority(), commonPath).toString();
   }
 
-  public static FileSelection create(DrillFileSystem fs, String parent, String path) throws IOException {
-    Path p = new Path(parent,removeLeadingSlash(path));
-    FileStatus[] status = fs.globStatus(p);
-    if (status == null || status.length == 0) {
+  public static FileSelection create(final DrillFileSystem fs, final String parent, final String path) throws IOException {
+    final Path combined = new Path(parent, removeLeadingSlash(path));
+    final FileStatus[] statuses = fs.globStatus(combined);
+    if (statuses == null) {
```

Comment:
if statuses is empty (but not null), we will call create below passing an empty statuses list and a null files list which will cause the precondition of the method to fail, right ?